clikit/src/main/java/org/adamalang/clikit/codegen/ArgumentsGen.java

```diff
@@ -0,0 +1,191 @@
+package org.adamalang.clikit.codegen;
+
+import org.adamalang.clikit.model.Argument;
+import org.adamalang.clikit.model.Command;
+import org.adamalang.clikit.model.Group;
+
+public class ArgumentsGen {
+    /**
+     * Generates the argument class, which is a collection of classes used for each command.
+     * Each class in the generated file represents args used in that command
+     **/
+    public static String generate(Group[] groups, Command[] mainCommands, String packageName) {
```

Comment:
So, this and other codegen comments are debatable. In this world, the class isn't the thing except a container for the static method. So, unlike others, it's OK to have the javadoc on the method. However, it's debatable which is why I'm just saying it's debatable rather something to do.